File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/application/port/outbound/bike/BikeSubscriptionRepository.java
package de.emaarco.example.application.port.outbound.bike;

import de.emaarco.example.domain.bike.BikeSubscription;
import de.emaarco.example.domain.bike.BikeSubscriptionId;

public interface BikeSubscriptionRepository {
    BikeSubscription save(BikeSubscription subscription);
    BikeSubscription find(BikeSubscriptionId id);
    BikeSubscription search(BikeSubscriptionId id);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/application/port/outbound/bike/BikeSubscriptionRepository.kt
package de.emaarco.example.application.port.outbound.bike

import de.emaarco.example.domain.bike.BikeSubscription
import de.emaarco.example.domain.bike.BikeSubscriptionId

interface BikeSubscriptionRepository {
    fun save(subscription: BikeSubscription): BikeSubscription
    fun find(id: BikeSubscriptionId): BikeSubscription
    fun search(id: BikeSubscriptionId): BikeSubscription?
}
